File: JavaScript/SubArrayDivision.js
function birthday(s, d, m) {
    let count=0,sum=0
    for(let i=0; i<s.length; i++){
        for(let j=i; j<m+i; j++){ sum+=s[j] }
        if(sum === d){ count++ }
        sum=0;
    }
    return count;

}

console.log(birthday([2, 2, 1, 3, 2], 4, 2))

File: Java/SubArrayDivision.java
public class SubArrayDivision {
    public static int birthday(int[] s, int d, int m) {
        int count = 0;
        for (int i = 0; i <= s.length - m; i++) {
            int sum = 0;
            for (int j = i; j < i + m; j++) {
                sum += s[j];
            }
            if (sum == d) {
                count++;
            }
        }
        return count;
    }
    
    public static void main(String[] args) {
        int[] s = {2, 2, 1, 3, 2};
        int d = 4;
        int m = 2;
        System.out.println(birthday(s, d, m));
    }
}
